### PR TITLE
CreateFolderCommand for Sharepoint/OneDrive

### PR DESCRIPTION
**Pull request by mereghost**
Related WP [OP#51856](https://community.openproject.org/projects/openproject/work_packages/51856)

The base implementation will only tackle the creation of a folder under the root of the Document Library.

### CODE REVIEW

File: modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/util.rb [hunk 1]
```diff
@@ -59,6 +59,14 @@ def using_user_token(storage, user, &)
         )
     end
 
+    def using_admin_token(storage)
+      oauth_client = storage.oauth_configuration.basic_rack_oauth_client
+
+      token = oauth_client.access_token!(scope: 'https://graph.microsoft.com/.default')
+
+      yield token
+    end
```
Comment (mereghost) [inline modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/util.rb]: This is a temporary patch before the rework of the ConnectionManager

File: modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/create_folder_command.rb [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+          def initialize(storage)
+            @storage = storage
+            @uri = storage.uri
+          end
+
+          # NOTE: This is currently creating a folder only on the root folder
+          def call(folder_path:)
+            Util.using_admin_token(@storage) do |token|
+              response = Net::HTTP.start(@uri.host, @uri.port, use_ssl: true) do |http|
```
Comment (Kharonus) [inline modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/create_folder_command.rb]: 🟡 Can you quick patch the new commands to use HTTPX?

File: modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/delete_folder_command.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          def call(location:)
+            Util.using_admin_token(@storage) do |token|
+              response = Net::HTTP.start(@uri.host, @uri.port, use_ssl: true) do |http|
+                http.delete(
+                  "/v1.0/drives/#{@storage.drive_id}/items/#{location}",
+                  Authorization: "Bearer #{token.access_token}"
+                )
+              end
+
+              ServiceResult.success(result: response.body)
```
Comment (Kharonus) [inline modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/delete_folder_command.rb]: 🟡 Is there truly no error handling necessary?

**Review by Kharonus — COMMENTED** [review]
I have two comments, which are both no blockers, but I think, those are "nice-to-haves".

File: modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/delete_folder_command.rb (hunk 3 above)
Comment (mereghost) [reply modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/delete_folder_command.rb]: Well... for now `delete` was more for testing purposes than actual usage. I can surely add it if you feel strongly about it.

Comment (Kharonus) [reply modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/delete_folder_command.rb]: it looks kinda unfinished. If you do not want to add it yet, I'd rather add a comment mentioning, that we need error handling once we have a use case.

Comment (mereghost) [reply modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/delete_folder_command.rb]: Added a test case for the most common case, which would be a 404.

**Review by Kharonus — APPROVED** [review]
LGTM

**Pull request merged**